### PR TITLE
Use _process_group in ReplicatedTensor and ShardedTensor.

### PR DESCRIPTION
**Pull request by pritamdamania87**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #77209
* #77191

As per comment in
https://github.com/pytorch/pytorch/pull/77191#discussion_r869641344, making
everthing consistent amongst these tensors.

Differential Revision: [D36300798](https://our.internmc.facebook.com/intern/diff/D36300798/)

### CODE REVIEW

File: torch/distributed/_shard/replicated_tensor.py [hunk 1]
```diff
@@ -33,15 +33,15 @@ class ReplicatedTensor(torch.Tensor):
     you wish to manually validate tensors are the same across ranks, use `validate()`.
 
     """
-    process_group: distributed_c10d.ProcessGroup
+    _process_group: distributed_c10d.ProcessGroup
 
     __slots__ = ["process_group"]
```
Comment (wanchaol) [inline torch/distributed/_shard/replicated_tensor.py:38]: nit: fix this `__slots__`?

**Review by wanchaol — APPROVED** [review]
looks good, thanks for fixing this!

**Review by fduwjj — APPROVED** [review]
LGTM

**Pull request closed without merging**